domain-http/interface/src/main/java/org/jboss/as/domain/http/server/cors/CorsUtil.java

```diff
@@ -53,10 +53,10 @@ public static boolean isCoreRequest(HeaderMap headers) {
      */
     public static String matchOrigin(HttpServerExchange exchange, Collection<String> allowedOrigins) throws Exception {
         HeaderMap headers = exchange.getRequestHeaders();
-        String origin = headers.getFirst(Headers.ORIGIN);
+        String origin = sanitizeDefaultPort(headers.getFirst(Headers.ORIGIN));
```

Comment:
I believe returning this sanitized value will violate the CORS spec. See Section 5.1 of http://www.w3.org/TR/cors/#access-control-allow-origin-response-header:

```The Access-Control-Allow-Origin header indicates whether a resource can be shared based by returning the value of the Origin request header, "*", or "null" in the response.```

The sanitized value is not the value of the Origin header.

I think you need to retain the origin value and only used the sanitized one in the matching checks.